### PR TITLE
Add load balancing support and user restriction for async jobs API

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

File: arangod/GeneralServer/AsyncJobManager.cpp [hunk 1]
```diff
@@ -125,16 +141,14 @@ void AsyncJobManager::deleteJobs() {
   auto it = _jobs.begin();
 
   while (it != _jobs.end()) {
-    GeneralResponse* response = (*it).second._response;
-
-    if (response != nullptr) {
-      delete response;
+    if (::authorized(it->second)) {
```
Comment (jsteemann) [inline arangod/GeneralServer/AsyncJobManager.cpp]: Is `it` increased in the `else` case too?
I think this is missing, so we will have an endless loop hee.

**Comment by jsteemann** [comment]
I have quickly fixed the issue I spotted during review. Can you retest with the current version and check if there are any issues now? Thanks!

**Comment by jsteemann** [comment]
@danielhlarkin : can you please change the header name in the documentation as well?
Please grep for the old header name in the docs and replace it. Thanks!

**Comment by danielhlarkin** [comment]
@jsteemann Will do. I still need to run/debug tests.

**Comment by danielhlarkin** [comment]
Ran the relevant load balancing tests manually and they pass locally. Letting Jenkins investigate now.
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/378/

**Pull request merged**